apisix/plugins/limit-conn/limit-conn-redis-cluster.lua

```diff
@@ -0,0 +1,139 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local redis_cluster     = require("apisix.utils.rediscluster")
+local core              = require("apisix.core")
+local assert            = assert
+local setmetatable      = setmetatable
+local math              = require "math"
+local floor             = math.floor
+local ngx_timer_at      = ngx.timer.at
+
+local _M = {version = 0.1}
+
+
+local mt = {
+    __index = _M
+}
+
+
+function _M.new(plugin_name, conf, max, burst, default_conn_delay)
+
+    local red_cli, err = redis_cluster.new(conf)
+    if not red_cli then
+        return nil, err
+    end
+    local self = {
+        conf = conf,
+        plugin_name = plugin_name,
+        burst = burst,
+        max = max + 0,    -- just to ensure the param is good
+        unit_delay = default_conn_delay,
+        red_cli = red_cli,
+    }
+    return setmetatable(self, mt)
+end
+
+
+function _M.incoming(self, key, commit)
+    local max = self.max
+    local red = self.red_cli
+    local conf = self.conf
+
+    self.committed = false
+
+    local prefix = conf.redis_prefix
```

Comment:
why is it necessary to use prefix

Reply:
Because the keys called by income are similar to 192.168.1.1route40, the prefix is added to distinguish plugins and facilitate key management, because users may use the same redis, and all of them are suitable for current limiting plugins.

Reply:
I think we can prefix the key with the plugin name so the key can become `limit-conn192.168.1route40`.

Reply:
like this:

https://github.com/shreemaan-abhishek/apisix/blob/0bea3dbf6b1b868ba35acdb2fd0205f2a352479d/apisix/plugins/limit-count/limit-count-redis.lua#L109

Reply:
OK, done.